students/mayc4t/lesson07/html_render.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+
+class TextWrapper:
+    """
+    A simple wrapper that creates a class with a render method
+    for simple text
+    """
+    def __init__(self, text):
+        self.text = text
+
+    def render(self, file_out, cur_ind='', new_line=True):
+        out_str = self.text
+        if new_line:
+            out_str = "{}\n".format(out_str)
+        file_out.write("{}{}".format(cur_ind, out_str))
+
+class Element:
+    tag = 'html'
+    ind = '  '
+    content =""
+
+
+    def __init__(self, content=None, **kwargs):
+        self.sub_elements = []
+        if content and isinstance(content, str):
+            self.sub_elements = [TextWrapper(content)]
+        elif content:
+            self.sub_elements = [content]
+        
+        self.attributes = ''
+        for key, value in kwargs.items():
+            self.attributes = '{} {}="{}"'.format(self.attributes, key, value)
+
+
+    def append( self, content ):
+        if isinstance(content, str):
+            self.sub_elements.append(TextWrapper(content))
+        else:
+            self.sub_elements.append(content)
+
+    def render(self, file_out, cur_ind='', new_line=True):
+        if new_line :
+            file_out.write("{}<{}{}>\n".format(cur_ind, self.tag, self.attributes))
+        else: 
+            file_out.write("{}<{}{}>".format(cur_ind, self.tag, self.attributes))
+        for idx_elm in self.sub_elements: 
+            idx_elm.render(file_out, '{}{}'.format(cur_ind, self.ind), new_line)
+        file_out.write("{}</{}>\n".format(cur_ind, self.tag))
+
+    def dump(self):
+        print('Element internals: tag={}, sub_elements={}'.format(self.tag, self.sub_elements))
```

Comment:
you could also define `__str__` method but I think you'll learn about that in Lesson 8 😉